packages/mobx/__tests__/v5/base/typescript-tests.ts

```diff
@@ -1887,6 +1887,18 @@ test("flow support throwing async generators", async () => {
     }
 })
 
+test("allow 'as const' for creating tuples for map.replace and friends", () => {
+    const map = mobx.observable.map<string, number>()
+    const srcValues = mobx.observable.array<string>()
+
+    const dispose = mobx.reaction(
```

Comment:
Yes, this how it'd look without `reaction`: 

```typescript
map.replace(Array.from(srcValues, (value, index) => [value, index] as const))
```

@Nokel81 Can you please update it?

Reply:
Sure, the reason I used reaction here is because in the replace case the intelisense is strong enough where the `as const` isn't needed whereas the reaction has some functions of distance.